File: 1749. Maximum Absolute Sum of Any Subarray/Maximum Absolute Sum of Any Subarray.py
nums = [2,-5,1,-4,3,-2]
min_prefix_sum = 0
max_prefix_sum = 0
prefix_sum = 0

for num in nums:
    prefix_sum += num

    min_prefix_sum = min(min_prefix_sum, prefix_sum)
    max_prefix_sum = max(max_prefix_sum, prefix_sum)

print(max_prefix_sum-min_prefix_sum)

File: 1749. Maximum Absolute Sum of Any Subarray/MaximumAbsoluteSumofAnySubarray.java
public class MaximumAbsoluteSumofAnySubarray {
    public int maxAbsoluteSum(int[] nums) {
        int max = 0, min = 0, sum = 0;
        for (int i: nums){
            sum += i;
            max = Math.max(sum, max);
            min = Math.min(min, sum);
        }
        return max-min;
    }
    public static void main(String[] args) {
        MaximumAbsoluteSumofAnySubarray mSubarray = new MaximumAbsoluteSumofAnySubarray();
        System.out.println(mSubarray.maxAbsoluteSum(new int[]{2,-5,1,-4,3,-2}));
    }
}
